haskell/test/Model/BaseDAO/Management.hs

```diff
@@ -0,0 +1,54 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Model.BaseDAO.Management
+  ( applyAcceptOwnership
+  , applyTransferOwnership
+  , applyCallCustom
+  ) where
+
+import Universum
+
+import Control.Monad.Except (throwError)
+import qualified Data.Map as M
+
+import qualified Michelson.Typed as T
+import Lorentz hiding (not, cast, checkSignature, get)
+
+import Ligo.BaseDAO.Types
+import Model.BaseDAO.Common
+import Model.BaseDAO.Types
+
+applyTransferOwnership :: ModelCall -> TransferOwnershipParam -> ModelT ()
+applyTransferOwnership mc param = do
+  selfAddr <- get <&> msSelfAddress
+  authorizeAdmin mc
+  let newOwner = arg #newOwner param
+
+  modifyStore \s ->
+    if selfAddr == newOwner then s { sAdmin = newOwner }
+    else s { sPendingOwner = newOwner}
+
+applyAcceptOwnership :: ModelCall -> ModelT ()
+applyAcceptOwnership mc = do
+  store <- getStore
+  if (store & sPendingOwner) == (mc & mcSender) then
+    modifyStore $ \s -> s { sAdmin = (mc & mcSender)}
+  else
+    throwError NOT_PENDING_ADMIN
+
+applyCallCustom :: ModelCall -> CallCustomParam -> ModelT ()
+applyCallCustom _ (epName, packedParam) = do
+  config <- getConfig
+  packedEp <-
+      case M.lookup epName (config & cCustomEntrypoints & unDynamic & T.unBigMap ) of
+        Just epCode -> pure epCode
+        Nothing -> throwError ENTRYPOINT_NOT_FOUND
+  case lUnpackValueRaw @('[(ByteString, FullStorage)] :-> '[([Operation], Storage)]) packedEp of
+    Right lambdaEp -> do
+      fullStorage <- get <&> msFullStorage
+      (ops, store) <- lambdaEp -$$? (packedParam, fullStorage)
```

Comment:
I don't think this is a good idea, given that this executes the lambda as-is it does the exact same thing as the interpreter, which defeats the purpose of testing custom entrypoints.

Here's what I think we should do: re-implement the lambdas for each DAO in the Haskell model ( taking `ByteString`s and using `ModelT`) and add a `Map` containing them in the `ModelState` (so that we can fill it differently based on which variant we are testing).
This way here instead of fetching and executing the lambda we can fetch and execute the function.

Reply:
I've implement the custom entryponts now, let me know if this is good enough.